xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-default/src/test/java/org/xwiki/security/authentication/internal/DefaultResetPasswordManagerTest.java

```diff
@@ -456,4 +463,151 @@ void resetPasswordNotDocumentReferenceUser() throws Exception
             () -> this.resetPasswordManager.resetPassword(otherUserReference, "some password"));
         assertEquals(exceptionMessage, resetPasswordException.getMessage());
     }
+
+    @Test
+    void isPasswordCompliantWithRegistrationRules()
```

Comment:
Imo this kind of test with a lot of identical assertions would be better as a parametrized test:
- clearer to read
- better error reporting as the test will cover all cases and not stop at the first error

Reply:
Agree I thought about it afterwards but was lazy :) I'll try to improve it